rasa/model_testing.py

```diff
@@ -128,7 +128,7 @@ async def test_core(
     model: Optional[Text] = None,
     stories: Optional[Text] = None,
     output: Text = DEFAULT_RESULTS_PATH,
-    additional_arguments: Optional[Dict] = None,
+    additional_arguments: Dict = {},
```

Comment:
We can't have mutable defaults in method params https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Reply:
Ah right! Good catch. 

What would be an alternative solution to https://github.com/RasaHQ/rasa/issues/10587 though? I can internally make a change so that internally the function goes: 

```python
if not additional_arguments:
    additional_arguments = {}
```

Would that suffice? The main gripe I have is an optional type that isn't optional.

Reply:
isn't that what's done [here](https://github.com/RasaHQ/rasa/blob/909374144524f16ec99025c151ce4e0d3eff662c/rasa/model_testing.py#L159)?

Reply:
Your link is from `test_core`, I'm concerned about `test_nlu` [here](https://github.com/RasaHQ/rasa/blob/909374144524f16ec99025c151ce4e0d3eff662c/rasa/model_testing.py#L188-L222).

Reply:
sorry, I was confused as you've changed the param in `test_core` 🙈 

You're right, we'd need to do what you said in `test_nlu` as well:

```python
if not additional_arguments:
    additional_arguments = {}
```

Reply:
Done!